ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -307,21 +311,28 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 // authenticate the incoming username/password and check if the resulting identity has
                                 // permission to run as the identity token principal.
                                 if (principal != null) {
-                                    SecurityIdentity authenticatedIdentity = this.authenticate(principal, (char[]) credential);
+                                    char[] password = (char[]) credential;
+                                    authenticationConfiguration = authenticationConfiguration.useName(principal.getName())
+                                            .usePassword(password);
+                                    SecurityIdentity authenticatedIdentity = this.authenticate(principal, password);
                                     identity = authenticatedIdentity.createRunAsIdentity(identityPrincipal.getName(), true);
                                 } else {
                                     // no TLS nor initial context token found - check if the anonymous identity has
                                     // permission to run as the identity principal.
                                     identity = this.securityDomain.getAnonymousSecurityIdentity().createRunAsIdentity(identityPrincipal.getName(), true);
                                 }
                             } else if (principal != null) {
+                                char[] password = (char[]) credential;
                                 // we have an initial context token containing a username/password pair.
-                                identity = this.authenticate(principal, (char[]) credential);
+                                authenticationConfiguration = authenticationConfiguration.useName(principal.getName())
+                                        .usePassword(password);
+                                identity = this.authenticate(principal, password);
                             }
                             final InterceptorContext interceptorContext = new InterceptorContext();
                             this.prepareInterceptorContext(op, params, interceptorContext);
                             try {
-                                retVal = identity.runAs((PrivilegedExceptionAction<Object>) () -> this.componentView.invoke(interceptorContext));
+                                final AuthenticationContext context = AuthenticationContext.empty().with(MatchRule.ALL, authenticationConfiguration);
```

Comment:
We are one step closer to being able to support it.

For forward authentication to work the security domain needs to be available to the AuthenticationConfiguration, however now it is associated with the deployments class loader we could have a default mode that uses the security domsin from the deployment.